packages/runtime/runtime-utils/src/generated/serializer.ejs

```diff
@@ -0,0 +1,238 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+<%_
+    // Perf: The motivation for this EJS template is share common code between 'recursivelyEncode()'
+    //       and 'recursivelyDecode()' without introducing a ~0.5x performance loss due to introducing
+    //       a callback (node 12 x64).
```

Comment:
The next ~50 lines are string processing helpers used to make the emitted *.ts file look natural.